File: app/src/main/java/com/example/aichat/SessionAssistantBindingStore.java
package com.example.aichat;

import android.content.Context;

import java.util.ArrayList;
import java.util.List;

public class SessionAssistantBindingStore {

    private final SessionAssistantBindingDao dao;

    public SessionAssistantBindingStore(Context context) {
        dao = AppDatabase.getInstance(context).sessionAssistantBindingDao();
    }

    public void bind(String sessionId, String assistantId) {
        if (sessionId == null || sessionId.isEmpty()) return;
        if (assistantId == null || assistantId.isEmpty()) {
            dao.delete(sessionId);
            return;
        }
        SessionAssistantBindingEntity entity = new SessionAssistantBindingEntity();
        entity.sessionId = sessionId;
        entity.assistantId = assistantId;
        dao.upsert(entity);
    }

    public String getAssistantId(String sessionId) {
        if (sessionId == null || sessionId.isEmpty()) return "";
        String result = dao.getAssistantId(sessionId);
        return result != null ? result : "";
    }

    public void remove(String sessionId) {
        if (sessionId == null || sessionId.isEmpty()) return;
        dao.delete(sessionId);
    }

    public boolean containsSession(String sessionId) {
        if (sessionId == null || sessionId.trim().isEmpty()) return false;
        return dao.contains(sessionId.trim()) > 0;
    }

    public List<String> getSessionIdsByAssistantId(String assistantId) {
        String target = assistantId != null ? assistantId.trim() : "";
        if (target.isEmpty()) return new ArrayList<>();
        List<String> result = dao.getSessionIdsByAssistantId(target);
        return result != null ? result : new ArrayList<>();
    }
}

File: app/src/main/java/com/example/aichat/SessionAssistantBindingStore.kt
package com.example.aichat

import android.content.Context

class SessionAssistantBindingStore(context: Context) {

    private val dao: SessionAssistantBindingDao = AppDatabase.getInstance(context).sessionAssistantBindingDao()

    fun bind(sessionId: String?, assistantId: String?) {
        if (sessionId.isNullOrEmpty()) return
        if (assistantId.isNullOrEmpty()) {
            dao.delete(sessionId)
            return
        }
        val entity = SessionAssistantBindingEntity()
        entity.sessionId = sessionId
        entity.assistantId = assistantId
        dao.upsert(entity)
        // 头像继承移到 ChatSessionActivity.initializeSessionOptionsFromAssistantOrGlobal,
        // 和其它 assistant options 一起初始化. 不能在 bind 阶段 save 空 options,
        // 否则 resolveChatOptions().has() 会误判为"已初始化", 跳过 assistant 设定继承.
    }

    fun getAssistantId(sessionId: String?): String {
        if (sessionId.isNullOrEmpty()) return ""
        val result = dao.getAssistantId(sessionId)
        return result ?: ""
    }

    fun remove(sessionId: String?) {
        if (sessionId.isNullOrEmpty()) return
        dao.delete(sessionId)
    }

    fun containsSession(sessionId: String?): Boolean {
        if (sessionId?.trim().isNullOrEmpty()) return false
        return dao.contains(sessionId!!.trim()) > 0
    }

    fun getSessionIdsByAssistantId(assistantId: String?): List<String> {
        val target = assistantId?.trim() ?: ""
        if (target.isEmpty()) return ArrayList()
        val result = dao.getSessionIdsByAssistantId(target)
        return result ?: ArrayList()
    }
}
